Sources/ParseLiveQuery/Client.swift

```diff
@@ -149,13 +150,18 @@ extension Client {
             )
             subscriptions.append(subscriptionRecord)
 
-            if socket == nil {
-                if !disconnected {
-                    reconnect()
-                }
-            } else {
-                sendOperationAsync(.Subscribe(requestId: subscriptionRecord.requestId, query: query))
+        if socket == nil || socket?.readyState != .OPEN{
```

Comment:
Looks like you don't need the second check and you could just reverse this control statement, meaning you can check whether the socket is open and send the subscribe event, otherwise check for not disconnected and stuff.
Would simplify the checking logic here.